packages/dataviews/src/view-table.js

```diff
@@ -306,16 +307,102 @@ function WithSeparators( { children } ) {
 		) );
 }
 
+function BulkSelectionCheckbox( { selection, onSelectionChange, data } ) {
+	const areAllSelected = selection && selection.length === data.length;
+	return (
+		<CheckboxControl
+			className="dataviews-table-selection-checkbox"
+			__nextHasNoMarginBottom
+			checked={ areAllSelected }
+			indeterminate={ ! areAllSelected && selection.length }
+			onChange={ () => {
+				if ( areAllSelected ) {
+					onSelectionChange( [] );
+				} else {
+					onSelectionChange( data );
+				}
+			} }
+			label={ areAllSelected ? __( 'Deselect all' ) : __( 'Select all' ) }
+		/>
+	);
+}
+
+function SingleSelectionCheckbox( {
+	selection,
+	onSelectionChange,
+	item,
+	labels,
+	data,
+	getItemId,
+	getItemTitle,
+} ) {
+	const id = getItemId?.( item );
```

Comment:
Is there any situation in which `getItemId` may not be declared? Do we need to cover against that case?

Reply:
We provide a default so we can remove the check and assume it is always declared.